### DISCUSSION TITLE
Most rides cannot use boosters whatsoever

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#21752] [Feature requests]
As someone who wants to make rides with more complicated behavior than default allows, I often want to include boosters or powered launches on coasters that do not allow them.

Ride types with no boosters usually have BoosterAcceleration set to 0, which means vehicles on that ride have no interaction with boosters. The workarounds to this are complicated and have tradeoffs. Currently, to circumvent the issues of zeroed BoosterAcceleration, the 'Allow Arbitrary Ride Type Changes' cheat is necessary to change the ride type of the actual ride to a type with a nonzero BoosterAcceleration. The vertical height offset of the ride is not always the same as the target ride, so users have to make sacrifices if they want the ride to operate as expected.

A discussion highlights the frustration with having zeroed BoosterAccelerations: https://github.com/OpenRCT2/OpenRCT2/discussions/18748. Half of this issue is solved by [Unify Speed](https://github.com/OpenRCT2/OpenRCT2/pull/16692) by having the vehicle use the track's RTD instead of the vehicle's RTD, but merges are still necessary with the current state of Unify Speed.

All ride types should have working boosters, so players do not have to deal with workarounds.

I do not believe it is as easy as setting BoosterAcceleration to nonzero values on all ride types. Some hacked rides rely on boosters being ineffective, so a switch must be added to choose the modern or legacy value. Unify Speed provides this framework with the least amount of overhead, so it is ideal to make any changes to BoosterAcceleration and PoweredLiftAcceleration in or directly following to prevent creating additional behavior modes.

**Comment by spacek531** [comment]
When https://github.com/OpenRCT2/OpenRCT2/pull/21956 is merged, all rides with zero BoosterSettings.BoosterAcceleration or BoosterSettings.LaunchedLiftHillAcceleration should have nonzero values, preferably ones that give a breadth of booster and launchedlifthill acceleration values to give hackers multiple values to choose from as setting these values per-track-piece is still not on the horizon.

**Comment by J-PIE-314** [comment]
#21658 is a similar request.

**Comment by ZeeMaji** [comment]
Requested note from the discord: RCT1 has different, unused boosters on the twister & hyper-twister coasters, that have a different acceleration than their RCTC/Open counterparts which breaks the rare parks that used them. They are roughly the same as the corkscrew boosters from what I recall.
